### PR TITLE
[JENKINS-36623] Test failing against Jenkins 1.651.3

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
[JENKINS-36623](https://issues.jenkins-ci.org/browse/JENKINS-36623)

@reviewbybees specially @oleg-nenashev

### CODE REVIEW

**Comment by armfergom** [comment]
Build is failing in the CI server, but it seems to be caused by a timeout. Does it build locally with core version defined in POM and a core version greater than 1.575?

**Comment by raul-arabaolaza** [comment]
@armfergom It builds and succesfully executes test with core version defined in the pom, it also succesfully executes the tests against 1.651.3 once built using the pom' s version. But, as expected,  it does not compile against 1.651.3.

**Comment by recena** [comment]
@raul-arabaolaza I recommend you to fill a new issue similar to [JENKINS-36646](https://issues.jenkins-ci.org/browse/JENKINS-36646)

**Comment by raul-arabaolaza** [comment]
Related issue [JENKINS-36654](https://issues.jenkins-ci.org/browse/JENKINS-36654)

**Comment by armfergom** [comment]
🐝

File: src/test/java/hudson/plugins/promoted_builds/KeepBuildForeverActionTest.java [hunk 1]
```diff
@@ -50,6 +56,11 @@ public void testDoesNotMarkBuildIfPromotionNotGoodEnough() throws Exception {
         promotionJob.getBuildSteps().add(new FixedResultBuilder(Result.FAILURE));
         promotionJob.getBuildSteps().add(new KeepBuildForeverAction());
 
+        // Make this test work with Jenkins 1.575+
+        for (ItemListener l : ItemListener.all()) {
```
Comment (oleg-nenashev) [inline src/test/java/hudson/plugins/promoted_builds/KeepBuildForeverActionTest.java]: Looks crappy. what is the reason of such call?

Comment (raul-arabaolaza) [inline src/test/java/hudson/plugins/promoted_builds/KeepBuildForeverActionTest.java]: @oleg-nenashev I took the idea from this comment https://github.com/jenkinsci/promoted-builds-plugin/blob/master/src/test/java/hudson/plugins/promoted_builds/PromotionProcessTest.java#L56 it seemed more suitable than using @LocalData. But any ideas/complaints are more than welcome

File: src/test/java/hudson/plugins/promoted_builds/conditions/inheritance/SelfPromotionInheritanceTest.java [hunk 2]
```diff
@@ -179,9 +180,15 @@ public void testPromotionEnvironmentShouldIncludeTargetParameters() throws Excep
         inheritanceProjectPair.getBase().addProperty(promotion);
 
         inheritanceProjectPair.getBase().addProperty(new ParametersDefinitionProperty(new StringParameterDefinition(paramName, "")));
+        inheritanceProjectPair.getDerived().addProperty(new ParametersDefinitionProperty(new StringParameterDefinition(paramName, "")));
```
Comment (oleg-nenashev) [inline src/test/java/hudson/plugins/promoted_builds/conditions/inheritance/SelfPromotionInheritanceTest.java]: Is it about SECURITY-170 again?

Comment (oleg-nenashev) [inline src/test/java/hudson/plugins/promoted_builds/conditions/inheritance/SelfPromotionInheritanceTest.java]: Smells like a bug in the in Inheritance Project plugin.
Makes sense to add TODO and to refer https://issues.jenkins-ci.org/browse/JENKINS-34831

File: src/test/java/hudson/plugins/promoted_builds/KeepBuildForeverActionTest.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/test/java/hudson/plugins/promoted_builds/KeepBuildForeverActionTest.java]: Please move this code to the utility method at least. too much duplications

**Comment by raul-arabaolaza** [comment]
@oleg-nenashev Comments addressed (I hope). Do you think is ok now??

File: src/main/java/hudson/plugins/promoted_builds/util/ItemListenerHelper.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package hudson.plugins.promoted_builds.util;
+
+import hudson.model.listeners.ItemListener;
+
+/**
+ * Utility class to call ArtifactArchiver.Migrator ItemListener to make this code compatible with jenkins 1.575+
+ */
+public class ItemListenerHelper {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/promoted_builds/util/ItemListenerHelper.java]: It should be in the test code, not in the production one

**Comment by oleg-nenashev** [comment]
@raul-arabaolaza 1 comment left

**Comment by raul-arabaolaza** [comment]
@oleg-nenashev Sorry for the ItemListenerHelper, fixed now. Was that the left comment??

**Comment by oleg-nenashev** [comment]
Maybe needs some refactoring, but OK for the test code

**Comment by recena** [comment]
LGTM, 🐝

**Pull request merged**